example/Makefile

```diff
@@ -1,9 +1,27 @@
-compile:
+tmp:
```

Comment:
This is similar to what we have in the `attestation-service-examples` repo.
- tmp for output to simplify jq commands
- check to make sure all the tools needed for the example are installed
- compile and x.wasm target to make sure x.wasm is rebuilt only if needed
- easier output parsing in run

Reply:
investigate if this is needed.  It add significant complexity that were previously unnecessary.